static/src/javascripts/bootstraps/standalone.commercial.ts

```diff
@@ -170,7 +171,18 @@ const bootCommercial = async (): Promise<void> => {
 		const promises = allModules.map((args) => loadModules(...args));
 
 		await Promise.all(promises).then(() => {
-			EventTimer.get().trigger('commercialModulesLoaded');
+			const eventTimer = EventTimer.get();
+			eventTimer.trigger('commercialModulesLoaded');
+			// how many ad slots are there on the page?
+			const adSlotsTotal = document.querySelectorAll(
+				`[id^="${AD_SLOT_ID_PREFIX}"]`,
+			).length;
```

Comment:
I was curious as to whether we already had this information recorded in the dfp env? If there's a one to one mapping between `Advert` and slot, would this produce a similar result? Or does it not capture the same information at this point in time?

```typescript
const numAdSlots = dfpEnv.adverts.length;
```

Reply:
I like the idea of collecting the information about how many slots there are as the slots are being inserted onto the page rather than doing a query selector later on. `dfpEnv.adverts` seems like the best place to retrieve the data from but at this point it isn't guaranteed to be populated with all of the elements it will eventually have. For fronts and liveblogs `dfpEnv.adverts.length` is consistently 0 when the commercial bundle is finished loading whereas for articles it seems to consistently match the number of slots found by the query selector approach. I think the discrepancy has to do with the fact that [prepare-googletag](https://github.com/guardian/frontend/blob/e8009589b79991f8fef544f46a5530d7ac402da3/static/src/javascripts/projects/commercial/modules/dfp/prepare-googletag.ts#L125) calls `fillAdvertSlots` asynchronously by pushing it on to `window.googletag.cmd` but I haven't worked out why there's a consistent difference between the different page types. 
If we want to use the `dfpEnv` approach I think we'll need to find another place to set the commercial metric property where `dfpEnv` will definitely be accurate. One possibility might be [loadAdvert](https://github.com/guardian/frontend/blob/b09acd49c20bdf7edbab12a5a390ff63909e1153/static/src/javascripts/projects/commercial/modules/dfp/load-advert.ts#L33-L34) but I think moving the logic there would mean calling `eventTimer.setProperty` multiple times which isn't ideal.
Very open to other suggestions if there are any!

Reply:
Thanks for the detailed thoughts Zeke!

I think for this PR the selector approach is fine (especially if the `dfpEnv` doesn't yield consistent results), as I suppose `document` does remain the source of truth for which slots are present at any given point in time.

There's probably a wider question as to where would be the best (and most consistent) place to hook into the `dfpEnv`. It does feel like we should be able to retrieve this information from there, or at least it should be easier to reason about what's in the `dfpEnv` at a given point.

Reply:
Yeah I think that makes sense to look into how we can find a place where all of the initial functions pushed onto `window.googletag.cmd` have been executed and the value of `dfpEnv.adverts` will be reliable.

Reply:
Going on a bit of a tangent - we should look at either removing `dfp-env` as unnecessary global state or at least replacing it with something that suits our needs better.

Once we have confidence in our e2e tests that might be a good point to start to try refactoring this.